pkg/hive/hiveconfig.go

```diff
@@ -0,0 +1,254 @@
+package hive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	hiveV1 "github.com/openshift/hive/apis/hive/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// HiveConfigBuilder provides struct for the HiveConfig object containing connection to
+// the cluster and the HiveConfig definitions.
+type HiveConfigBuilder struct {
+	Definition *hiveV1.HiveConfig
+	Object     *hiveV1.HiveConfig
+	errorMsg   string
+	apiClient  *clients.Settings
```

Comment:
can you add 
```
import (
   runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
)
```
And change this to
```suggestion
	apiClient  runtimeClient.Client
```